### PR TITLE
ENG-139 [FEAT]: Github OAuth for Dub

### PR DESCRIPTION
**Pull request by versecafe**
## Added GitHub OAuth

Related Issue https://github.com/dubinc/dub/issues/538

### Register Screen
image

### Logged In
image

> Also added to the log in screen and the env vars needed are in .env.example along with links to next-auth docs for setup.

### CODE REVIEW

**Comment by versecafe** [comment]
Closes https://github.com/dubinc/dub/issues/538

File: apps/web/lib/auth/options.ts [hunk 1]
```diff
@@ -196,6 +202,24 @@ export const authOptions: NextAuthOptions = {
             },
           });
         }
+      } else if (account?.provider === "github") {
+        console.log("github user!");
```
Comment (itssubhodiproy) [inline apps/web/lib/auth/options.ts]:
```suggestion
```

**Review by subh-cs — CHANGES_REQUESTED** [review]
Great stuff @versecafe  🚀

**Review by steven-tey — APPROVED** [review]

**Comment by steven-tey** [comment]
Incredible work on this PR, thank you so much @versecafe! 🙏

**Pull request merged**